### PR TITLE
Change 3D split to 2D split

### PR DESCRIPTION
**Pull request by gmegan**
Resolves Issue #23 

Added individual options for the two result teams specifically to support splitting multiple times with some intermediary teams being non-collective.

### CODE REVIEW

File: content/shmem_team_split_2d.tex [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+non-negative and all \acp{PE} in the parent team must pass the same value for
+\VAR{xrange}. None of the parameters need to reside in symmetric memory.
+
+Since the split may result in a 2D space with more points than there are members of
+the parent team, there may be a final, incomplete row of the 2D mapping of the parent
+team. This means that the resulting X-axis teams may vary in size by up to 1 \ac{PE},
+and that there may be one resulting Y-axis team of smaller size than all of the other
+Y-axis teams.
+
+The following grid shows the twelve teams that would result from splitting a parent team
```
Comment (nspark) [inline content/shmem_team_split_2d.tex]: I think that this example (and maybe the following paragraph) should be in the _Notes_ section, not the behavioral description.

File: content/shmem_team_split_2d.tex [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+is passed as the parent team handle. In that case, the program will abort with
+an informative error message. If an invalid team handle is passed to the
+routine, the behavior is undefined.
+
+The following options can be supplied during team split to restrict
+team functions and enable performance optimizations.  When using a given
+team, the application must comply with the requirements of all options
+set on that team; otherwise, the behavior is undefined.
+No options are enabled on \LibHandleRef{SHMEM\_TEAM\_WORLD} or on other
+predefined \openshmem teams.
```
Comment (nspark) [inline content/shmem_team_split_2d.tex]: I think we need to say something to the effect of "all PEs in the resultant team must specify the same options". We have to be precise in the text here because, as noted earlier, the resultant X-axis and Y-axis teams don't need to have the same options at each other. It also seems like it should be valid that, for example, the Y-axis team where _y_ is zero might specify `NOCOLLECTIVE` while Y-axis teams where _y_ is not zero might not.

In other words, for all resultant teams from a call to `split_2d`, all PEs in the team must pass the same options for that team.

**Review by nspark — CHANGES_REQUESTED** [review]

File: content/shmem_team_split_2d.tex (hunk 2 above)
Comment (gmegan) [reply content/shmem_team_split_2d.tex]: Ah yes, that does make more sense. I will be more clear on this and maybe add a sentence to the example in the notes section.

Comment (gmegan) [reply content/shmem_team_split_2d.tex]: I am realizing this change needs to be made to the split_strided as well. I will put that in a different PR.

**Review by nspark — APPROVED** [review]
